### PR TITLE
In 151030885 user can see contact log

### PR DESCRIPTION
**Pull request by NateOSI**
### Pivotal Story

- [**INVESTIGATIONS** User can see a list of existing contacts in the contacts log #151030885](https://www.pivotaltracker.com/story/show/151030885)

### Purpose
**As an** investigator
**I need** to see a consolidated record of all exisitng contacts for an investigation
**So that** I can quickly reference the right contact for details of what has transpired so far, and to quickly see what work still needs to be done before I can close the investigation

# Context
https://www.pivotaltracker.com/epic/show/3710639

### CODE REVIEW

File: app/javascript/investigations/ContactLogRow.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  return (
+    <tr>
+      <td>{startedAt}</td>
+      <td>{people.join(', ')}</td>
+      <td>
+        <div>{method}</div>
+        <div>({status})</div>
+      </td>
+      <td>
+        <div>{nWords(note, MAX_NUMBER_OF_WORDS)}</div>
```
Comment (saraOSI) [inline app/javascript/investigations/ContactLogRow.jsx]: Since the contact log selector is doing all of the other presentation formatting work, why not truncate the note there, too?

File: app/javascript/utils/textFormatter.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export const nWords = (text, numberOfWords) => {
```
Comment (saraOSI) [inline app/javascript/utils/textFormatter.js]: 👍

File: config/application.rb [hunk 3]
```diff
@@ -29,14 +29,12 @@ class Application < Rails::Application # :nodoc:
...
-    base_path = ENV.fetch('BASE_PATH', '')
-
     config.intake = {
       api_url: ENV.fetch('API_URL', nil),
       authentication_base_url: ENV.fetch('AUTHENTICATION_URL', ''),
       authentication_login_url: authentication_login_url,
-      base_path: base_path,
-      dashboard_url: Rails.root.join(base_path),
+      base_path: ENV.fetch('BASE_PATH', ''),
+      dashboard_url: Rails.root,
```
Comment (saraOSI) [inline config/application.rb]: Why the change for dashboard here?

File: lib/tasks/spec.rake [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 def host_env_string
-  'REDIS_HOST=$(docker-machine ip intake) REDIS_PORT=6379 API_URL=http://api'
+  'REDIS_HOST=redis_host REDIS_PORT=6379 API_URL=http://api FERB_API_URL=https://ferb_api'
```
Comment (saraOSI) [inline lib/tasks/spec.rake]: 👍

File: config/application.rb (hunk 3 above)
Comment (NateOSI) [reply config/application.rb]: Dashboard never includes the base_path since its a separate application i.e. http://dashboard_url/intake/...

File: app/javascript/investigations/ContactLogRow.jsx (hunk 1 above)
Comment (NateOSI) [reply app/javascript/investigations/ContactLogRow.jsx]: I believe product plans to expand notes in the future with a user action.

File: config/application.rb (hunk 3 above)
Comment (saraOSI) [reply config/application.rb]: Gotcha.

File: app/javascript/investigations/ContactLog.jsx [hunk 5]
```diff
@@ -1,24 +1,60 @@
...
+                const props = {
+                  id,
+                  investigationId,
+                  startedAt,
+                  people,
+                  method,
+                  status,
+                  note,
+                }
+                return (<ContactLogRow key={index} {...props} />)
```
Comment (jgedeon-osi) [inline app/javascript/investigations/ContactLog.jsx]: Maybe we shouldn't reconstruct this in a var called props but pass them directly.
